enhancements/kubelet/virtualization-higher-workload-density.md

```diff
@@ -0,0 +1,502 @@
+
+* **Phase 1** - OpenShift Virtualization will provide an out-of-tree
+  solution (WASP) to enable higher workload density and swap-based eviction mechanisms.
+* **Phase 2** - WASP will include swap-based eviction mechanisms.
+* **Phase 3** - OpenShift Virtualization will transition to [Kubernetes SWAP] (in-tree).
+  OpenShift will [allow](#swap-ga-for-cnv-users-only) SWAP to be configured only if OpenShift Virtualization is installed on the cluster.
+  In this phase, WASP deprecation will start.
+* **Phase 4** - OpenShift will GA SWAP for every user, even if OpenShift Virtualization
+  is not installed on the cluster. In this phase WASP will be removed, machine-level configuration
+  will be managed by Machine Config Operator.
+
+### Workflow Description
+
+**cluster administrator** (Cluster Admin) is a human user responsible
+for deploying and administering a cluster.
+
+**virtual machine owner** (VM Owner) is a human user operating a
+virtual machine in a cluster.
+
+#### Workflow: Configuring higher workload density
+
+1. The cluster admin is deploying a bare-metal OpenShift cluster
+2. The cluster admin is deploying the OpenShift Virtualization Operator
+3. The cluster admin is deploying the WASP Agent according to the
+   [documentation](https://github.com/openshift-virtualization/wasp-agent/blob/main/docs/configuration.md)
+4. The cluster admin is configuring OpenShift Virtualization for higher
+   workload density via:
+
+   a. the OpenShift Virtualization Console "Settings" page 
+   b. [or `HCO` API](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/docs/cluster-configuration.md#configure-higher-workload-density)
+
+The cluster is now set up for higher workload density.
+
+In phase 3, deploying the WASP agent will not be needed.
+
+#### Workflow: Leveraging higher workload density
+
+1. The VM Owner is creating a regular virtual machine and is launching it. The VM owner must not specify memory requests in the VM spec, but only the guest memory size.
+
+### API Extensions
+
+#### Phase 3
+In phase 3 WASP will be replaced by kubernetes swap, which should be GA by then. With that being said it's requried to update the worker kubelet configuration with `memorySwap.swapBehavior`.
+Following is a KubeletConfig example:
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: KubeletConfig
+metadata:
+  name: custom-config
+spec:
+  machineConfigPoolSelector:
+    matchLabels:
+      custom-kubelet: enabled
+  kubeletConfig:
+    failSwapOn: false
+    memorySwap:
+      swapBehavior: LimitedSwap
+```
+More info about kubelet swap API can be found [here](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2400-node-swap/README.md#api-changes)
+
+#### Phase 4
+In previous phases the machine-level confgiuration was deployed manually using machine configs. The transition to phase 4
+will require re-deployment of the configuration by the operator.
```

Comment:
"the operator" this is not very clear. I think we should spell out what will be taken over by the swap operator.

Reply:
if there are open questions, call them out

Reply:
I've added content to the open question section regarding the swap operator